test/spec/modules/oneVideoBidAdapter_spec.js

```diff
@@ -0,0 +1,116 @@
+import { expect } from 'chai';
+import { spec } from 'modules/oneVideoBidAdapter';
+import * as utils from 'src/utils';
+
+describe('OneVideoBidAdapter', () => {
+  let bidRequest;
+
+  beforeEach(() => {
+    bidRequest = {
+      bidder: 'oneVideo',
+      sizes: [640, 480],
+      bidId: '30b3efwfwe1e',
+      params: {
+        video: {
+          playerWidth: 480,
+          playerHeight: 640,
+          mimes: ['video/mp4', 'application/javascript'],
+          protocols: [2, 5],
+          api: [2],
+          position: 1,
+          delivery: [2]
+        },
+        site: {
+          id: 1,
+          page: 'https://news.yahoo.com/portfolios',
+          referrer: 'http://www.yahoo.com'
+        },
+        pubId: 'brxd'
+      }
+    };
+  });
+
+  describe('spec.isBidRequestValid', () => {
+    it('should return true when the required params are passed', () => {
+      expect(spec.isBidRequestValid(bidRequest)).to.equal(true);
+    });
+
+    it('should return false when the "video" param is missing', () => {
+      bidRequest.params = {
+        pubId: 'brxd'
+      };
+      expect(spec.isBidRequestValid(bidRequest)).to.equal(false);
+    });
+
+    it('should return false when the "pubId" param is missing', () => {
+      bidRequest.params = {
+        video: {
+          playerWidth: 480,
+          playerHeight: 640,
+          mimes: ['video/mp4', 'application/javascript'],
+          protocols: [2, 5],
+          api: [2],
+          position: 1,
+          delivery: [2]
+        }
+      };
+      expect(spec.isBidRequestValid(bidRequest)).to.equal(false);
+    });
+
+    it('should return false when no bid params are passed', () => {
+      bidRequest.params = {};
+      expect(spec.isBidRequestValid(bidRequest)).to.equal(false);
+    });
+  });
+
+  describe('spec.buildRequests', () => {
+    it('should create a POST request for every bid', () => {
+      const requests = spec.buildRequests([ bidRequest ]);
+      expect(requests[0].method).to.equal('POST');
+      expect(requests[0].url).to.equal(location.protocol + spec.ENDPOINT + bidRequest.params.pubId);
+    });
+
+    it('should attach the bid request object', () => {
+      const requests = spec.buildRequests([ bidRequest ]);
+      expect(requests[0].bidRequest).to.equal(bidRequest);
+    });
+  });
```

Comment:
should also be validating that the data payload looks correct

Reply:
done